contrib/internal/sqltest/sqltest.go

```diff
@@ -266,26 +422,37 @@ func testExec(cfg *Config) func(*testing.T) {
 			for k, v := range cfg.ExpectTags {
 				assert.Equal(v, span.Tag(k), "Value mismatch on tag %s", k)
 			}
+			// Since SQLServer runs execute statements by doing a prepare first, the expected comment
+			// excludes dynamic tags which can only be injected on non-prepared statements
+			expectedCommentTags = map[string]tagExpectation{
+				"ddsn":  {MustBeSet: true, ExpectedValue: "test-service"},
+				"ddsp":  {MustBeSet: false},
+				"ddsid": {MustBeSet: false},
+				"ddtid": {MustBeSet: false},
+			}
 		} else {
 			assert.Len(spans, 5)
 		}
 
 		var span mocktracer.Span
 		for _, s := range spans {
-			if s.OperationName() == cfg.ExpectName && s.Tag(ext.ResourceName) == query {
+			rn, _ := s.Tag(ext.ResourceName).(string)
+			if s.OperationName() == cfg.ExpectName && strings.HasSuffix(rn, query) {
 				span = s
 			}
 		}
-		assert.NotNil(span, "span not found")
+		require.NotNil(t, span, "span not found")
```

Comment:
Updated this to require because otherwise a nil span causes a panic below and aborts all tests. This makes it easier to understand what's actually broken.